tsl/src/gapfill/interpolate.c

```diff
@@ -125,14 +125,11 @@ gapfill_fetch_sample(GapFillState *state, GapFillInterpolateColumnState *column,
 	value = heap_getattr(&tuple, 1, tupdesc, &sample->isnull);
 	if (!sample->isnull)
 	{
-		sample->time = value;
+		sample->time = gapfill_datum_get_internal(value, state->gapfill_typid);
```

Comment:
why not use `ts_time_value_to_internal` from utils? Is it correct to just get the int64 value from timestamptz types? I thought there is a timezone offset that needs to be added in these cases.

Reply:
ts_time_value_to_internal does not handle timestamp correctly, also i'd rather use the postgres internal format instead of some arbitrary other one. This saves us from having to convert back and forth between the formats whenever we need to use it.

Reply:
timestamptz is always UTC so no timezone handling is needed, unlike timestamp which is in local timezone.

Reply:
when  HAVE_INT64_TIMESTAMP is set, timestamp types are int64 which is always true on PG10+ and is the default on PG9.6, gapfill code won't compile without this set.